deps/rabbit/src/rabbit_amqqueue.erl

```diff
@@ -504,7 +504,7 @@ maybe_migrate(ByNode, MaxQueuesDesired, [N | Nodes]) ->
         [{_, Q, false} = Queue | Queues] = All when length(All) > MaxQueuesDesired ->
             Name = amqqueue:get_name(Q),
             Module = rebalance_module(Q),
-            Candidates = Module:get_replicas(Q) -- [N],
+            Candidates = rabbit_maintenance:filter_out_drained_nodes_consistent_read(Module:get_replicas(Q) -- [N]),
```

Comment:
Do we really need a consistent read here? It involves a cluster-wide query. We will do this for every queue and it is really unlikely that the list of nodes under maintenance will change in the middle of the rebalancing operation (in practice).

Reply:
You're right. Changed to dirty